src/tasks/tasks.guard.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Observable } from 'rxjs';
+
+// this was my work with guards. Nothing fansy. 
+@Injectable()
+export class TasksGuard implements CanActivate {
+    canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
```

Comment:
This was my attempt to work with guards. So I created one and used it in controller.